### PR TITLE
oshmem: add pkg-config files

### PR DESCRIPTION
**Pull request by ggouaillardet**
Refs open-mpi/ompi#4730

Signed-off-by: Gilles Gouaillardet <gilles@rist.or.jp>

### CODE REVIEW

File: config/oshmem_config_files.m4 [hunk 1]
```diff
@@ -14,18 +14,22 @@
 
 AC_DEFUN([OSHMEM_CONFIG_FILES],[
     AC_CONFIG_FILES([
-    oshmem/Makefile
-    oshmem/include/Makefile
-    oshmem/shmem/c/Makefile
-    oshmem/shmem/c/profile/Makefile
+        oshmem/Makefile
+        oshmem/include/Makefile
```
Comment (artpol84) [inline config/oshmem_config_files.m4]: Are these indentations necessary?

**Comment by artpol84** [comment]
@Di0gen can you double check if this is ok?

**Comment by hppritcha** [comment]
@jladd-mlnx please review, otherwise this won't make 4.0

**Comment by jsquyres** [comment]
@ggouaillardet Do you foresee progressing this PR?

**Comment by ggouaillardet** [comment]
I restored the original indentation and this PR is ready for merge

@jladd-mlnx can you please review it ?

**Comment by AboorvaDevarajan** [comment]
Can one of the admins verify this patch?

**Comment by awlauria** [comment]
Ping. Do we want this PR?

**Comment by gpaulsen** [comment]
Probably yes, we want this, but I'm not familiar enough to review.
@jladd-mlnx ?

**Comment by awlauria** [comment]
@jladd-mlnx re-ping. Please review or assign someone to review. Thanks.

**Comment by jladd-mlnx** [comment]
@amaslenn can you take a look, please.

File: oshmem/tools/wrappers/oshmem-c.pc.in [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+exec_prefix=@exec_prefix@
+includedir=@includedir@
+libdir=@libdir@
+pkgincludedir=@opalincludedir@
+#
+# Note that -lopen-pal and all the EXTRA_LIBS are only necessary when
+# static linking (they're pulled in by libopen-rte.so's implicit
+# dependencies), so only list these in Libs.private.
+#
+Libs: -L${libdir} @OMPI_PKG_CONFIG_LDFLAGS@ -loshmem -l@OMPI_LIBMPI_NAME@
```
Comment (amaslenn) [inline oshmem/tools/wrappers/oshmem-c.pc.in]: (minor) Does it make sense to create a `OMPI_LIBOSHMEM_NAME` to replace hardcoded `oshmem`?

**Comment by awlauria** [comment]
Re-ping. 5.0 branching is targeted for April 30th. If you want this in for 5.0, please try to get it in master by then. Thanks!

**Comment by lanl-ompi** [comment]
Can one of the admins verify this patch?

File: oshmem/tools/wrappers/oshmem-c.pc.in (hunk 2 above)
Comment (jsquyres) [reply oshmem/tools/wrappers/oshmem-c.pc.in:21]: If this is the only issue left on this PR, I'd say that it's ready / done / can be merged.

@ggouaillardet @amaslenn ?

Comment (amaslenn) [reply oshmem/tools/wrappers/oshmem-c.pc.in:21]: Nothing apart from this one from my side.

Comment (jsquyres) [reply oshmem/tools/wrappers/oshmem-c.pc.in:21]: @ggouaillardet If you still want this one, I'd say you're clear to merge.

**Comment by gpaulsen** [comment]
@ggouaillardet @open-mpi/ucx - What's the fate of this PR?

**Review by awlauria — APPROVED** [review]

**Comment by awlauria** [comment]
approving based on review comments from @amaslenn and @jsquyres .

Looks like it needs a retrigger of CI. Hm.

**Comment by jsquyres** [comment]
This is from 2018.  A lot has happened on master since then; you might want to check that this PR is on-par with the OMPI package files that are on master.

**Comment by awlauria** [comment]
bot:retest

**Comment by awlauria** [comment]
@amaslenn @mellanox-github can you verify this is still a good patch?

**Comment by amaslenn** [comment]
> @amaslenn @mellanox-github can you verify this is still a good patch and does not need updating?

@yosefe could you please take a look or assign someone?

**Review by yosefe — APPROVED** [review]

**Comment by jsquyres** [comment]
I told you this stuff was from 2018 and it was probably wrong.

Did anyone actually test this before it was merged?

I just looked at the merge commit email that went by, and I see that it is definitely, positively wrong.

This needs to be reverted.  If you're going to actually merge something into master -- especially from a PR from 3 years ago -- you need to actually review it and test it.  Don't just rubber stamp it and merge it in without thinking.

👎  👎  👎

**Pull request merged**